substrate/frame/message-queue/src/lib.rs

```diff
@@ -1447,7 +1448,23 @@ impl<T: Config> Pallet<T> {
 		use ProcessMessageError::*;
 		let prev_consumed = meter.consumed();
 
-		match T::MessageProcessor::process_message(message, origin.clone(), meter, &mut id) {
+		let transaction = storage::with_transaction(
```

Comment:
Yes this seems to go into the right direction. Need a test to actually verify that it works though 👍

Reply:
Ok. i will work on the test